### DISCUSSION TITLE
Copilot4Eclipse initialization fails after updating to latest Eclipse LSP4E version

### DISCUSSION THREAD

**Discussion by fbuechler** [Genuitec/Copilot4Eclipse-community#33] [Help] [answered]
After updating Eclipse to org.eclipse.lsp4e 0.18.14.202410030435, the Copilot plugin is stuck in the status "Waiting for Copilot4Eclipse initialization".
The error log repeatedly shows "Init failed - Copilot LS is not active" but no further details.
Reverting to org.eclipse.lsp4e 0.18.13.202409050855 solves the problem.

![grafik](https://github.com/user-attachments/assets/8c33686d-50f6-44d3-b431-9d28e849d76c)

**Comment by wayneparrott** [comment]
Thanks for sharing. We will investigate and follow up asap.

Wayne
Copilot4Eclipse Team

**Comment by wayneparrott** [comment]
I would like to replicate your experience. Can you share how you installed org.eclipse.lsp4e 0.18.14.202410030435?
That version doesn't appear to be part of the the 2024-09 or even the 2024-12M1 environments I've looked at. 

W

**Comment by wayneparrott** [comment]
Also, any interesting error log entries just after startup?

**Comment by fbuechler** [comment]
Thanks for looking into this!

The update to 0.18.14.202410030435 came up during the Eclipse's regular check for updates. It seems to come from https://download.eclipse.org/lsp4e/releases/latest/ which is enabled for me as "Available Software Site". I can't remember whether I've added this site manually or if it was there by default. Rolling back and disabling the site works for me.

The only other error logged on startup is the following:
```
bundle org.apache.felix.scr:2.2.12 (3669) Circular reference detected trying to get service {org.eclipse.m2e.core.project.IMavenProjectChangedListener}={service.id=267, service.bundleid=3287, service.scope=bundle, osgi.ds.satisfying.condition.target=(osgi.condition.id=true), component.name=org.eclipse.m2e.core.internal.project.WorkspaceStateWriter, component.id=18}
 stack of references: ServiceReference: {org.eclipse.m2e.core.project.IMavenProjectChangedListener}={service.id=267, service.bundleid=3287, service.scope=bundle, osgi.ds.satisfying.condition.target=(osgi.condition.id=true), component.name=org.eclipse.m2e.core.internal.project.WorkspaceStateWriter, component.id=18}
    Dependency: DependencyManager: Component [Component: org.eclipse.m2e.core.internal.project.registry.MavenProjectManager (21)] reference [MavenProjectChangedListener]
ServiceReference: {org.eclipse.m2e.core.project.IMavenProjectRegistry}={service.id=260, service.bundleid=3287, service.scope=bundle, osgi.ds.satisfying.condition.target=(osgi.condition.id=true), component.name=org.eclipse.m2e.core.internal.project.registry.MavenProjectManager, component.id=21}
    Dependency: DependencyManager: Component [Component: org.eclipse.m2e.core.embedder.MavenModelManager (6)] reference [projectManager]
    Dependency: DependencyManager: Component [Component: org.eclipse.m2e.core.embedder.MavenModelManager (6)] reference [projectManager]
ServiceReference: {org.eclipse.m2e.core.internal.project.registry.ProjectRegistryManager}={service.id=258, service.bundleid=3287, service.scope=bundle, osgi.ds.satisfying.condition.target=(osgi.condition.id=true), component.name=org.eclipse.m2e.core.internal.project.registry.ProjectRegistryManager, component.id=23}
```

**Comment by jcompagner** [comment]
yes i have the same thing now, 
Not sure what happens here because i don't think i updated eclipse the last past days, and i have an install that did work (but after restart it doesn't)
And all new workspaces i tries get the same error.

**Reply by jcompagner** [reply]
this is really weird i checked my install/updates and the latest was on 11 oktober..

And yesterday and even today (that was still started) my main eclipse workspace did work fine, but after restart it doesn't anymore

So now any eclipse on any workspace doesn't start and the worst thing is i really need to restart it now and then because constantly a node process (with a conhost.exe) is started every few minutes or so, that takes more and more memory..

**Comment by Philipp0205** [comment]
The same happened to me as well. The plugin is stuck at  "Waiting for Copilot4Eclipse initialization". 
I can confirm everything @fbuechler said.


My seesion data is: 

```
eclipse.buildId=4.30.5.202407311447
java.version=21.0.2
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
Framework arguments:  -Xms2G -Xmx80G
Command-line arguments:  -os linux -ws gtk -arch x86_64 -data file:/...../ -Xms2G -Xmx80G
```

The update happened on the 08.10 from 
Language Server Protocol client for Eclipse IDE (Incubation)        0.18.14.202410030435 to
Language Server Protocol client for Eclipse IDE (Incubation)        0.18.13.202409050855

**Comment by wayneparrott** [comment]
@jcompagner @Philipp0205 
Thx for sharing this info. Investigating now. 

What Eclipse distro and additional plugins do you have installed?

Will update you as soon as I know more. In the meantime is it possible for you to roll back to an earlier LSP version?

**Comment by Philipp0205** [comment]
@wayneparrott Thanks for looking into that! 

Update: I used the eclipse installation history to revert all installed software to the state it was on the 8th of October where Copilot broke. Now it works again. In my case, the only two parts of the IDE updated.

EWC / DTA Projects Support and Yourkit Integration	1.5.27.202410291422
and 
Language Server Protocol client for Eclipse IDE (Incubation)	0.18.14.202410030435

I think it's unlikely, that Yourkit is the reason, so it was LSP after all. 

If anybody wants to revert, the last working LSP version (0.18.13...) can be found under the update url https://download.eclipse.org/lsp4e/releases/0.26.5/

**Reply by Philipp0205** [reply]
I edited my original post.

**Comment by wayneparrott** [comment]
@Philipp0205 this info is very helpful. I will try to carve out some time and replicate the issue.

W

**Comment by jcompagner** [comment]
@wayneparrott  i couldn't revert, Eclipse didn't let me do that (just reverting that lspe) also reverting to an older snapshot completely didn't work because of copilot p2 site complaining it couldn't find that older release..

So i did a fresh install of Eclipse 2024.09 and installed the stuff that i need and then it worked, now i have lsp4e 0.18.12.xx

I was already afraid it would go wrong again because i also install WWD from the snapshot site and that has already that newer release, but it didn't upgrade it when i installed WWD.

**Reply by Philipp0205** [reply]
I also had difficulty only reverting lsp4e. Did you Help -> About -> Installation history -> and revert from there? That worked for me.

**Reply by jcompagner** [reply]
no that failed because i needed to revert the quite a bit (it was not just lsp4e) and then i needed to revert also a copilot install (that was part of it ) and that failed because i think they only have the latest in the site, so then you cant revert)

**Comment by brenosilver** [comment]
Same problem using a brand new installation of STS 4.26.0

eclipse.buildId=4.33.0.202410162246
java.version=21.0.5
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en_US
Framework arguments:  -product org.springframework.boot.ide.branding.sts4
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.springframework.boot.ide.branding.sts4 -data file:/C:/Users/{user}/Documents/workspace-spring-tool-suite-4-4.26.0.RELEASE/

**Comment by rishirajmusa** [comment]
I was facing the same issue so I uninstalled the current eclipse and installed the previous version and it worked.
Below is the GitHub link for previous versions.
https://github.com/spring-projects/sts4/wiki/Previous-Versions

**Comment by wayneparrott** [comment]
We are investigating this issue and hope to have an update that is compatible with the latest lsp4e plugin very soon.

W

**Comment by wayneparrott** [comment]
Update - I was able to isolate and fix this issue yesterday. The upcoming Copilot4Eclipse 1.3.2 will include this fix. Thx for all the good input.

**Comment by wayneparrott** [comment, marked as answer]
Today we rolled out Copilot4Eclipse 1.3.2 that includes broader compatibility including the supported versions of LSP4E. Please update to it and let us know if you observe any issues. We appreciate all of the input and research you have provided to help get this issue resolved.

W

**Comment by yogeshgowdabn-w** [comment]
![screebn](https://github.com/user-attachments/assets/ce8077c4-d679-4e2f-8ae0-21f82590dc99)
please resolve this issue with eclipse

**Reply by brianvfernandes** [reply]
Yogesh, please open a new discussion for your issue. Please mention the version and distribution of Eclipse you're using as well as the version of Copilot4Eclipse. Also include the complete stack trace of the error in your report. Thanks!
